gossip-protocol-spec.md

```diff
@@ -102,15 +109,37 @@ struct SomeType {
     y: u16,
 }
 ```
-In the first case the serialized object of `CompressionType` enum will only contain 4-byte header with the discriminant value set to the selected variant (`0 = GZip`, `1 = Bzip2`). In the latter case apart from the header the serialized data will contain additional bytes according to which variant was selected: 
+In the first case, the serialized object of `CompressionType` enum will only contain a 4-byte header with the discriminant value set to the selected variant (`0 = GZip`, `1 = Bzip2`). In the latter case apart from the header the serialized data will contain additional bytes according to which variant was selected: 
 * `Variant1`: 8 bytes
 * `Variant2`: 6 bytes (the sum of `x` and `y` fields of `SomeType` struct)
 
-A special care need to be taken when deserializing such enum as according to the selected variant number of following data bytes may be different.
+Special care needs to be taken when deserializing such enum as according to the selected variant number of following data bytes may be different.
+
+### PushMessage
+It is sent by nodes who want to share information with others. Nodes gather data from their `crds` and send push messages to their peers periodically.
 
+Nodes receiving the messages check them for duplication, insert them into their `crds` in case the received value doesn't exist or is updated and transmit them further to their peers.
```

Comment:
Maybe change a little to something like: 
A node receiving a set of PushMessages will: 
* checks for duplicate `CrdsValue` and drop them
* insert new `CrdsValue`s into the `crds`
* transmit newly inserted `CrdsValue`s to their peers via PushMessage 

It's just a little vague sometimes. In the line highlighted, "them" refers to both PushMessages and CrdsValues in the same sentence. Generally, I would avoid using "it", "them", "they" etc since these terms are vague unless it is absolutely clear what "it" or "they" refers to.